sdks/java/core/src/main/java/org/apache/beam/sdk/util/Credentials.java

```diff
@@ -132,15 +135,32 @@ public static Credential getCredential(GcpOptions options)
    * Loads OAuth2 credential from a local file.
    */
   private static Credential getCredentialFromFile(
-      String keyFile, String accountId, Collection<String> scopes)
+          @Nullable String keyFile, @Nullable String accountName, Collection<String> scopes)
       throws IOException, GeneralSecurityException {
-    GoogleCredential credential = new GoogleCredential.Builder()
+
+    if (keyFile == null) {
+      throw new IOException("If accountName is given, also supply a keyFile");
+    }
```

Comment:
Hi Alex, I'm clearly missing something here. It looks like from the code at L141 that `keyFile` MUST be non-null to enter this function. But the change you made at L111 makes it okay to call this function if `keyFile` is `null`.

Should L111 maybe just be `if (keyFile != null)` ? Then we can remove `@Nullable` from the parameter here.